// @aws-sdk/client-timestream-write/dist-cjs/TimestreamWrite.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimestreamWrite = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const CreateBatchLoadTaskCommand_1 = require("./commands/CreateBatchLoadTaskCommand");
const CreateDatabaseCommand_1 = require("./commands/CreateDatabaseCommand");
const CreateTableCommand_1 = require("./commands/CreateTableCommand");
const DeleteDatabaseCommand_1 = require("./commands/DeleteDatabaseCommand");
const DeleteTableCommand_1 = require("./commands/DeleteTableCommand");
const DescribeBatchLoadTaskCommand_1 = require("./commands/DescribeBatchLoadTaskCommand");
const DescribeDatabaseCommand_1 = require("./commands/DescribeDatabaseCommand");
const DescribeEndpointsCommand_1 = require("./commands/DescribeEndpointsCommand");
const DescribeTableCommand_1 = require("./commands/DescribeTableCommand");
const ListBatchLoadTasksCommand_1 = require("./commands/ListBatchLoadTasksCommand");
const ListDatabasesCommand_1 = require("./commands/ListDatabasesCommand");
const ListTablesCommand_1 = require("./commands/ListTablesCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const ResumeBatchLoadTaskCommand_1 = require("./commands/ResumeBatchLoadTaskCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateDatabaseCommand_1 = require("./commands/UpdateDatabaseCommand");
const UpdateTableCommand_1 = require("./commands/UpdateTableCommand");
const WriteRecordsCommand_1 = require("./commands/WriteRecordsCommand");
const TimestreamWriteClient_1 = require("./TimestreamWriteClient");
const commands = {
    CreateBatchLoadTaskCommand: CreateBatchLoadTaskCommand_1.CreateBatchLoadTaskCommand,
    CreateDatabaseCommand: CreateDatabaseCommand_1.CreateDatabaseCommand,
    CreateTableCommand: CreateTableCommand_1.CreateTableCommand,
    DeleteDatabaseCommand: DeleteDatabaseCommand_1.DeleteDatabaseCommand,
    DeleteTableCommand: DeleteTableCommand_1.DeleteTableCommand,
    DescribeBatchLoadTaskCommand: DescribeBatchLoadTaskCommand_1.DescribeBatchLoadTaskCommand,
    DescribeDatabaseCommand: DescribeDatabaseCommand_1.DescribeDatabaseCommand,
    DescribeEndpointsCommand: DescribeEndpointsCommand_1.DescribeEndpointsCommand,
    DescribeTableCommand: DescribeTableCommand_1.DescribeTableCommand,
    ListBatchLoadTasksCommand: ListBatchLoadTasksCommand_1.ListBatchLoadTasksCommand,
    ListDatabasesCommand: ListDatabasesCommand_1.ListDatabasesCommand,
    ListTablesCommand: ListTablesCommand_1.ListTablesCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    ResumeBatchLoadTaskCommand: ResumeBatchLoadTaskCommand_1.ResumeBatchLoadTaskCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateDatabaseCommand: UpdateDatabaseCommand_1.UpdateDatabaseCommand,
    UpdateTableCommand: UpdateTableCommand_1.UpdateTableCommand,
    WriteRecordsCommand: WriteRecordsCommand_1.WriteRecordsCommand,
};
class TimestreamWrite extends TimestreamWriteClient_1.TimestreamWriteClient {
}
exports.TimestreamWrite = TimestreamWrite;
(0, smithy_client_1.createAggregatedClient)(commands, TimestreamWrite);
