// @aws-sdk/client-timestream-write/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateBatchLoadTaskRequestFilterSensitiveLog = exports.RejectedRecordsException = exports.DimensionValueType = exports.TableStatus = exports.PartitionKeyType = exports.PartitionKeyEnforcementLevel = exports.ValidationException = exports.ThrottlingException = exports.ServiceQuotaExceededException = exports.ResourceNotFoundException = exports.InvalidEndpointException = exports.InternalServerException = exports.ConflictException = exports.S3EncryptionOption = exports.TimeUnit = exports.ScalarMeasureValueType = exports.MeasureValueType = exports.BatchLoadStatus = exports.BatchLoadDataFormat = exports.AccessDeniedException = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const TimestreamWriteServiceException_1 = require("./TimestreamWriteServiceException");
class AccessDeniedException extends TimestreamWriteServiceException_1.TimestreamWriteServiceException {
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        this.name = "AccessDeniedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
    }
}
exports.AccessDeniedException = AccessDeniedException;
exports.BatchLoadDataFormat = {
    CSV: "CSV",
};
exports.BatchLoadStatus = {
    CREATED: "CREATED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PENDING_RESUME: "PENDING_RESUME",
    PROGRESS_STOPPED: "PROGRESS_STOPPED",
    SUCCEEDED: "SUCCEEDED",
};
exports.MeasureValueType = {
    BIGINT: "BIGINT",
    BOOLEAN: "BOOLEAN",
    DOUBLE: "DOUBLE",
    MULTI: "MULTI",
    TIMESTAMP: "TIMESTAMP",
    VARCHAR: "VARCHAR",
};
exports.ScalarMeasureValueType = {
    BIGINT: "BIGINT",
    BOOLEAN: "BOOLEAN",
    DOUBLE: "DOUBLE",
    TIMESTAMP: "TIMESTAMP",
    VARCHAR: "VARCHAR",
};
exports.TimeUnit = {
    MICROSECONDS: "MICROSECONDS",
    MILLISECONDS: "MILLISECONDS",
    NANOSECONDS: "NANOSECONDS",
    SECONDS: "SECONDS",
};
exports.S3EncryptionOption = {
    SSE_KMS: "SSE_KMS",
    SSE_S3: "SSE_S3",
};
class ConflictException extends TimestreamWriteServiceException_1.TimestreamWriteServiceException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
}
exports.ConflictException = ConflictException;
class InternalServerException extends TimestreamWriteServiceException_1.TimestreamWriteServiceException {
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
exports.InternalServerException = InternalServerException;
class InvalidEndpointException extends TimestreamWriteServiceException_1.TimestreamWriteServiceException {
    constructor(opts) {
        super({
            name: "InvalidEndpointException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidEndpointException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidEndpointException.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidEndpointException = InvalidEndpointException;
class ResourceNotFoundException extends TimestreamWriteServiceException_1.TimestreamWriteServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class ServiceQuotaExceededException extends TimestreamWriteServiceException_1.TimestreamWriteServiceException {
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "ServiceQuotaExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.ServiceQuotaExceededException = ServiceQuotaExceededException;
class ThrottlingException extends TimestreamWriteServiceException_1.TimestreamWriteServiceException {
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        this.name = "ThrottlingException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.Message = opts.Message;
    }
}
exports.ThrottlingException = ThrottlingException;
class ValidationException extends TimestreamWriteServiceException_1.TimestreamWriteServiceException {
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ValidationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
    }
}
exports.ValidationException = ValidationException;
exports.PartitionKeyEnforcementLevel = {
    OPTIONAL: "OPTIONAL",
    REQUIRED: "REQUIRED",
};
exports.PartitionKeyType = {
    DIMENSION: "DIMENSION",
    MEASURE: "MEASURE",
};
exports.TableStatus = {
    ACTIVE: "ACTIVE",
    DELETING: "DELETING",
    RESTORING: "RESTORING",
};
exports.DimensionValueType = {
    VARCHAR: "VARCHAR",
};
class RejectedRecordsException extends TimestreamWriteServiceException_1.TimestreamWriteServiceException {
    constructor(opts) {
        super({
            name: "RejectedRecordsException",
            $fault: "client",
            ...opts,
        });
        this.name = "RejectedRecordsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, RejectedRecordsException.prototype);
        this.Message = opts.Message;
        this.RejectedRecords = opts.RejectedRecords;
    }
}
exports.RejectedRecordsException = RejectedRecordsException;
const CreateBatchLoadTaskRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ClientToken && { ClientToken: smithy_client_1.SENSITIVE_STRING }),
});
exports.CreateBatchLoadTaskRequestFilterSensitiveLog = CreateBatchLoadTaskRequestFilterSensitiveLog;
